xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTask.java

```diff
@@ -607,6 +608,21 @@ public TaskPipeline<ModifySpaceQuery> createPipeline() {
     }
   }
 
+  public static class ModifySubscriptionQuery extends FeatureTask<ModifySubscriptionEvent, ModifySubscriptionQuery> {
```

Comment:
This class can extend the Task class directly as the implementation is not related to features at all. Common parts (e.g. resolveSpace) could be pulled up into Task.